addon/data-load.js

```diff
@@ -913,12 +913,34 @@ export class Editor extends React.Component {
         ["\"", "\""],
       ]);
       const closeChar = openToCloseChar.get(e.key);
-      model.editor.setRangeText(e.key, selectionStart, selectionStart, "end");
-      model.editor.setRangeText(closeChar, selectionEnd + 1, selectionEnd + 1, "preserve");
+      // if quote (or any other start char) before and quote (or any other corresponding end char) right after it do not add quote (or the corresponding end char) but just move cursor after the it
+      if (selectionStart === selectionEnd && model.editor.value.substring(selectionStart - 1, selectionStart) == e.key && model.editor.value.substring(selectionEnd, selectionEnd + 1) == closeChar) {

```

Comment:
selectionstart can be begining of string so it start with -1...
same issue with selectionEnd which can be at end. You can get index out of range.

Reply:
it work ony for quote because it is same character but if i understand well, when you add [ and you already have [ before it and ] after cursor, you will move after ].... seems odd to me.